hibernate-core/src/main/java/org/hibernate/sql/results/graph/collection/internal/AbstractCollectionInitializer.java

```diff
@@ -183,4 +184,9 @@ public CollectionKey resolveCollectionKey(RowProcessingState rowProcessingState)
 	public void finishUpRow(RowProcessingState rowProcessingState) {
 		collectionKey = null;
 	}
+
+	@Override
+	public EntityInitializer asEntityInitializer() {
+		return ( EntityInitializer ) this;
```

Comment:
doesn't need a cast?

Reply:
Intellij IDEA warning me again.

Reply:
I quickly checked this case: the warnings IDEA is showing you are correct because this type is NOT an `EntityInitializer`.
Seems something is wrong here, and that's why I'd prefer to not have the casts as they will point to real issues.